### PR TITLE
Cube can be generated

### PR DESCRIPTION
**Pull request by Xipit**
#1 

- Cube has a fixed Datastructure
- Cubes can be Instantiated as GameObjects

### CODE REVIEW

File: Assets/Scripts/CubeScript.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        prototypeCube = new Cube(prototypeCubePosition, Quaternion.Euler(prototypeCubeRotation), dimension);
+
+        if (fieldPrefab)
+        {
+            prototypeCube.instantiateObjects(fieldPrefab);
+        }
+    }
+
+    // Update is called once per frame
+    void Update()
```
Comment (Semeseen) [inline Assets/Scripts/CubeScript.cs:31]: Update() function can be removed. It isn't used

File: Assets/Scripts/Extensions.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+// Extensions to existing classes
+// https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods
+namespace Extensions
+{
+    public static class Extensions
```
Comment (Semeseen) [inline Assets/Scripts/Extensions.cs:9]: do we need this class?

File: Assets/Scripts/CubeScript.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+{
+    public GameObject fieldPrefab;
+
+    public Cube prototypeCube;
+
+    public Vector3 dimensionAsVector; // only for editing in Unity Editor
+    public Dimension3D dimension;
+
+    public Vector3 prototypeCubePosition;
+    public Vector3 prototypeCubeRotation; // doesnt work correctly TODO fix
```
Comment (Semeseen) [inline Assets/Scripts/CubeScript.cs:15]: Will there be a bug item or do we leave it like this for the next weeks?

File: Assets/Plane.prefab
Comment (Semeseen) [inline Assets/Plane.prefab:1]: It looks like we don't need this GameObject anymore. Am I right?

Comment (Xipit) [reply Assets/Plane.prefab:1]: The 2 prefab objects are for testing purpose and will be removed once the 3D assets will be ready.

**Pull request merged**